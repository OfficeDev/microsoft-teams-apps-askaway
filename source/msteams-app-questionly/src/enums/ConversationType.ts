/**
 * Define value for conversation type in teams app.
 */
export enum ConversationType {
    Channel = 'channel',
}
