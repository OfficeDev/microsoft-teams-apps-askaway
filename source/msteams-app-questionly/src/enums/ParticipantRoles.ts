/**
 * Define value for participant role in a meeting.
 */
export enum ParticipantRoles {
    Presenter = 'Presenter',
    Organizer = 'Organizer',
    Attendee = 'Attendee',
}
