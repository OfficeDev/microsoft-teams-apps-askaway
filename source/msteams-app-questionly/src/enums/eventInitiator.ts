/**
 * Define value for different event initiators.
 */
export enum EventInitiator {
    MainCard = 'MainCard',
    RestApi = 'RestApi',
}
