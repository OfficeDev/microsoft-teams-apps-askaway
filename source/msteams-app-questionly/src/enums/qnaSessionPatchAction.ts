/**
 * Patch actions supported for qna session.
 */
export enum QnaSessionPatchAction {
    End = 'end',
}
