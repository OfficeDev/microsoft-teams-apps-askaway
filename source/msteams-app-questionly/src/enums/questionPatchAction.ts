/**
 * Patch actions supported for question.
 */
export enum QuestionPatchAction {
    Upvote = 'upvote',
    Downvote = 'downvote',
    MarkAnswered = 'markAnswered',
}
