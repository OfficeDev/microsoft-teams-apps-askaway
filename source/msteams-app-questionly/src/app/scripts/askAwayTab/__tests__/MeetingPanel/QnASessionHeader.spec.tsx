/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { Flex, Text, FlexItem, Menu, ThemePrepared } from '@fluentui/react-northstar';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import { QnASessionHeader } from '../../MeetingPanel/QnASessionHeader';
import { ParticipantRoles } from '../../../../../enums/ParticipantRoles';
import { themeMock } from '../mocks/themes';

configure({ adapter: new enzymeAdapterReact16() });

describe('QnASessionHeader Component', () => {
    const title = 'some-title';
    let t;
    let onClickRefreshSession;
    let onClickEndSession;
    let theme;

    beforeAll(() => {
        t = jest.fn();
        onClickRefreshSession = jest.fn();
        onClickEndSession = jest.fn();
        theme = themeMock;
    });

    it('should match the snapshot', () => {
        const userRole = ParticipantRoles.Presenter;
        const showToolBar = true;
        const wrapper = shallow(
            <QnASessionHeader t={t} title={title} theme={theme} onClickRefreshSession={onClickRefreshSession} onClickEndSession={onClickEndSession} userRole={userRole} showToolBar={showToolBar} />
        );
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the QnASessionHeader when showToolBar is false', () => {
        const userRole = ParticipantRoles.Presenter;
        const showToolBar = false;
        const component = shallow(
            <QnASessionHeader t={t} title={title} theme={theme} onClickRefreshSession={onClickRefreshSession} onClickEndSession={onClickEndSession} userRole={userRole} showToolBar={showToolBar} />
        );

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(1);
        expect(component.find(FlexItem)).toHaveLength(0);
    });

    it('should render the QnASessionHeader with menu when showToolBar is true', () => {
        const userRole = ParticipantRoles.Presenter;
        const showToolBar = true;
        const component = shallow(
            <QnASessionHeader t={t} title={title} theme={theme} onClickRefreshSession={onClickRefreshSession} onClickEndSession={onClickEndSession} userRole={userRole} showToolBar={showToolBar} />
        );

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(1);
        expect(component.find(FlexItem)).toHaveLength(1);
        expect(component.find(Menu)).toHaveLength(1);
    });
});
