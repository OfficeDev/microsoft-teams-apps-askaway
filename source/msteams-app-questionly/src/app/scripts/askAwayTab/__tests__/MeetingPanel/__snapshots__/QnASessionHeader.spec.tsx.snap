// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QnASessionHeader Component should match the snapshot 1`] = `
<Flex
  className="qna-header"
  gap="gap.large"
>
  <Text
    as="span"
    className="truncated-title"
    content="some-title"
    size="large"
    title="some-title"
  />
  <FlexItem
    push={true}
  >
    <div
      className="menuHeader"
    >
      <Menu
        accessibility={[Function]}
        aria-label="Compose Editor"
        as="ul"
        defaultActiveIndex={0}
        iconOnly={true}
        items={
          Array [
            Object {
              "aria-label": "More options",
              "icon": <MoreIcon
                outline={false}
                styles={
                  Object {
                    "color": "#fff",
                  }
                }
              />,
              "indicator": false,
              "key": "moreOptions",
              "menu": Object {
                "items": Array [
                  Object {
                    "content": undefined,
                    "icon": <RetryIcon
                      outline={true}
                    />,
                    "key": "Refresh session",
                    "onClick": [Function],
                  },
                  Object {
                    "content": undefined,
                    "icon": <LeaveIcon
                      outline={true}
                    />,
                    "key": "End session",
                    "onClick": [Function],
                  },
                ],
              },
            },
          ]
        }
      />
    </div>
  </FlexItem>
</Flex>
`;
